src/Files.App/Views/LayoutModes/StandardLayoutMode.cs

```diff
@@ -248,23 +250,47 @@ protected async void RenameTextBox_KeyDown(object sender, KeyRoutedEventArgs e)
 					e.Handled = true;
 					break;
 				case VirtualKey.Up:
-					textBox.SelectionStart = 0;
+					if (isShiftPressed)
+					{
```

Comment:
```suggestion
					if (!isShiftPressed)
					     textBox.SelectionStart = 0;
```
I think this should be enough

Reply:
Nope it isn't

Reply:
I can do some more testing on my side, but it should be enough to just return early is shift is pressed.

Reply:
Yep you might try. But I have already tried doing such things and it mostly didn't work so I had written full behavior for every case.